subprojects/language-native/src/integTest/groovy/org/gradle/language/AbstractNativeLanguageIncrementalBuildIntegrationTest.groovy

```diff
@@ -176,8 +176,13 @@ abstract class AbstractNativeLanguageIncrementalBuildIntegrationTest extends Abs
         executedAndNotSkipped ":linkHelloSharedLibrary"
         executedAndNotSkipped ":helloSharedLibrary"
         skipped mainCompileTask
-        executedAndNotSkipped ":linkMainExecutable"
-        executedAndNotSkipped ":mainExecutable"
+        if (linkTimeFileShouldChange()) {
```

Comment:
I started down this path (new exported symbol) but the problem is that one of the tests is to change a compiler flag in the library and check that everything rebuilds.  If we added a new exported symbol in that case, we'd be opening the door for a false positive on that test (i.e. re-compile/re-link gets triggered by the changed source rather than the compile flag).  We could wrap the new symbol in an `ifdef` or something, but then I was concerned we might muddy another test.  In the end, it just felt like too much of a workaround.

I think I prefer just checking fewer outcomes.